cluster/addons/metrics-server/resource-reader.yaml

```diff
@@ -23,6 +23,7 @@ rules:
   verbs:
   - get
   - list
+  - update
```

Comment:
Note that it is done differently in heapster. There is a Role "system:pod-nanny" which is bound here: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/cluster-monitoring/heapster-rbac.yaml#L51. Here is analogous config for nanny: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/metrics-server/auth-reader.yaml. I think it's the better way, since it separates permissions for different components, e.g. if nanny needs additional permissions, you will have to change only one config.

Reply:
I considered doing this similarly but IIUC the role is binded to the same service account as the other Heapster roles. I think it will make sense to have such model if nanny runs as a separate pod. Otherwise it's just confusing. WDYT? Do I understand it correctly?

Reply:
Yes, this will grant the same permissions for all containers. The benefit I see is the separation of configuration, i.e. if we have multiple components using pod nanny, we only need to manage one configuration file. wrt. readability, it's more readable in some ways and less in others. In heapster it's easier to see which permissions are needed for which purpose, on the other hand there is no explicit list of all permissions in one place. IMO it's better to use "system:pod-nanny" role.